### PR TITLE
[TE] rootcause poc - aggregation loading, heatmap, and frontend rollup

### PR DESCRIPTION
**Pull request by apucher**
* add aggregation loading, caching (this needs to be refactored into a service)

* add rootcause-heatmap component for translation

* implement frontend data rollup capabilities

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/components/heatmap-chart/component.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  },
+
+  _makeSum(dimNameObj, funcExtract) {
+    if (!dimNameObj) {
+      return 0;
+    }
+    return Object.values(dimNameObj).reduce((agg, x) => agg + funcExtract(x), 0);
+  },
+
+  _makeSumOther(dimNameObj, topk) {
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/pods/components/heatmap-chart/component.js]: What are `dimNamObj` and `topk`? Is there a way we can rename these variables to make them more meaningful?

File: thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs [hunk 2]
```diff
@@ -63,5 +63,7 @@ urns
 
 <h3>This is the heatmap.</h3>
 {{rootcause-heatmap
-  mode="change"
-}}
+  aggregates=aggregates
+  currentUrns=heatmapCurrentUrns
+  current2baseline=heatmapCurrent2Baseline
+}}
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs]: Nitpick: add a new line after this

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/template.hbs [hunk 3]
```diff
@@ -1,17 +1,9 @@
-<button {{action "setMode" "change"}}>change</button> |
-<button {{action "setMode" "contributionDiff"}}>contribution diff</button> |
-<button {{action "setMode" "contributionToDiff"}}>contribution to diff</button>
+urns
+{{#each currentUrns as |urn|}}
+  | <button {{action "setCurrentUrn" urn}}>{{urn}}</button>
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/template.hbs]: fyi, for straight foward click action that sets a property to a new value, we can use the `mut` helper here and a custom actions is not needed anymore. 
something like: 
```
onclick={{action (mut urn) urn}}
```
https://emberjs.com/api/ember/2.15/classes/Ember.Templates.helpers/methods/mut?anchor=mut

File: thirdeye/thirdeye-frontend/app/pods/components/heatmap-chart/component.js (hunk 1 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/components/heatmap-chart/component.js]: hmm not really without blowing this helper out of proportion. "topk" is the set of topk dimension names. "dimNameObj" only makes sense in connection with the code that calls this helper.

File: thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs (hunk 2 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/rootcause/template.hbs]: fixed

File: thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/template.hbs (hunk 3 above)
Comment (apucher) [reply thirdeye/thirdeye-frontend/app/pods/components/rootcause-heatmap/template.hbs]: yes thanks. fixed

**Pull request merged**